clicache/src/DistributedSystem.cpp

```diff
@@ -470,14 +470,15 @@ namespace Apache
         return instance;
       }
 
-      DistributedSystem::DistributedSystem(std::unique_ptr<native::DistributedSystem> nativeptr)
+      DistributedSystem::DistributedSystem(native::DistributedSystem nativeDistributedSystem)
```

Comment:
We don't want copies. Copy constructor should be deleted. This should be r-value reference.